### PR TITLE
Add secret manager block

### PR DESCRIPTION
**Pull request by ahuang11**
Adds a SecretsManager block, inheriting from the SecretsBlock interface.

### Screenshots

image

### Checklist

- [x] References any related issue by including "Closes #<Issue Number>" or "Closes <Issue URL>".
  - If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect-aws/issues/new/choose) first.
- [x] Includes tests or only affects documentation.
- [x] Passes `pre-commit` checks.
  - Run `pre-commit install && pre-commit run --all` locally for formatting and linting.
- [x] Includes screenshots of documentation updates.
  - Run `mkdocs serve` view documentation locally.
- [x] Summarizes PR's changes in [CHANGELOG.md](https://github.com/PrefectHQ/prefect-aws/blob/main/CHANGELOG.md)

### CODE REVIEW

File: prefect_aws/credentials.py [hunk 1]
```diff
@@ -231,3 +241,12 @@ def get_s3_client(self) -> S3Client:
...
         return self.get_client(client_type=ClientType.S3)
+
+    def get_secrets_manager_client(self) -> SecretsManagerClient:
+        """
+        Gets an authenticated Secrets Manager client.
+
+        Returns:
+            An authenticated Secrets Manager client.
+        """
+        return self.get_client(client_type=ClientType.SECRETS_MANAGER)
```
Comment (desertaxle) [inline prefect_aws/credentials.py:252]: I don't think that MinIO implements a secret manager interface, just a storage one, so `MinIOCredentials` shouldn't be able to provide a secrets manager client.
```suggestion

    def get_secrets_manager_client(self) -> SecretsManagerClient:
        """
        Gets an authenticated Secrets Manager client.

        Returns:
            An authenticated Secrets Manager client.
        """
        return self.get_client(client_type=ClientType.SECRETS_MANAGER)
```

File: prefect_aws/secrets_manager.py [hunk 2]
```diff
@@ -350,3 +352,152 @@ def example_delete_secret_with_recovery_window():
     except ClientError:
         logger.exception("Unable to delete secret %s", secret_name)
         raise
+
+
+class SecretsManager(SecretBlock):
+    """
+    Block that represents a resource that can store and retrieve secrets.
```
Comment (desertaxle) [inline prefect_aws/secrets_manager.py:359]:
```suggestion
    Manages a secret in AWS's Secrets Manager.
```

File: prefect_aws/secrets_manager.py [hunk 3]
```diff
@@ -350,3 +352,152 @@ def example_delete_secret_with_recovery_window():
...
+
+    Attributes:
+        aws_credentials: The credentials to use for authentication with AWS.
+        secret_name: The name of the secret.
+    """
+
+    aws_credentials: Union[AwsCredentials, MinIOCredentials]
```
Comment (desertaxle) [inline prefect_aws/secrets_manager.py:366]: Related to the comment above about MinIO, this should only accept `AwsCredentials`.
```suggestion
    aws_credentials: AwsCredentials
```

File: prefect_aws/secrets_manager.py [hunk 4]
```diff
@@ -350,3 +352,152 @@ def example_delete_secret_with_recovery_window():
...
+        Examples:
+            Write some secret data.
+            ```python
+            secrets_manager = SecretsManager.load("MY_BLOCK")
+            secrets_manager.write_secret(b"my_secret_data")
+            ```
+        """
+        client = self.aws_credentials.get_secrets_manager_client()
+        try:
+            response = await run_sync_in_worker_thread(
```
Comment (desertaxle) [inline prefect_aws/secrets_manager.py:435]: Does `run_sync_in_worker_thread` work outside of a flow?

Comment (ahuang11) [reply prefect_aws/secrets_manager.py:435]: Yes it works:
```
    call = partial(__fn, *args, **kwargs)
    return await anyio.to_thread.run_sync(
        call, cancellable=True, limiter=get_thread_limiter()
    )
```

Comment (ahuang11) [reply prefect_aws/secrets_manager.py:435]:
```
import anyio
def test():
    print("Hello")
await anyio.to_thread.run_sync(test, cancellable=True)
```

**Review by desertaxle — APPROVED** [review]
LGTM!

**Pull request merged**